complete-firebase-authentication-react-tutorial.md

```diff
@@ -372,209 +360,385 @@ const config = {
+const config = { ... };
 
-export {
-  auth,
-  firebase,
-};
+class Firebase {
+  constructor() {
+    app.initializeApp(config);
+
+    this.auth = app.auth();
+  }
+
+  // *** Auth API ***
+
+  doCreateUserWithEmailAndPassword = (email, password) =>
+    this.auth.createUserWithEmailAndPassword(email, password);
+
+  doSignInWithEmailAndPassword = (email, password) =>
+    this.auth.signInWithEmailAndPassword(email, password);
+}
+
+export default Firebase;
 {{< /highlight >}}
 
-That way, consumers (React components in our case) should be only allowed to access the *index.js* file as the sole interface to the entire Firebase module (*src/firebase/*),  and should not access the auth or firebase files directly.
+These endpoints are called asynchronously, and they will need to be resolved later, as well as error handling. For instance, it is not possible to sign in a user who is not signed up yet since the Firebase API would return an error.
 
-{{% chapter_header "Sign Up with React and Firebase" "react-firebase-sign-up" %}}
+You don't need to pass any argument to the sign out function because the Firebase knows about the currently authenticated user. If no user is authenticated, nothing will happen this function is called.
+
+{{< highlight javascript "hl_lines=21" >}}
+import app from 'firebase/app';
+import 'firebase/auth';
+
+const config = { ... };
+
+class Firebase {
+  constructor() {
+    app.initializeApp(config);
+
+    this.auth = app.auth();
+  }
+
+  // *** Auth API ***
+
+  doCreateUserWithEmailAndPassword = (email, password) =>
+    this.auth.createUserWithEmailAndPassword(email, password);
+
+  doSignInWithEmailAndPassword = (email, password) =>
+    this.auth.signInWithEmailAndPassword(email, password);
+
+  doSignOut = () => this.auth.signOut();
+}
+
+export default Firebase;
+{{< /highlight >}}
+
+There are two more authentication methods to reset and change a password for an authenticated user:
+
+{{< highlight javascript "hl_lines=23 25 26" >}}
+import app from 'firebase/app';
+import 'firebase/auth';
+
+const config = { ... };
+
+class Firebase {
+  constructor() {
+    app.initializeApp(config);
+
+    this.auth = app.auth();
+  }
+
+  // *** Auth API ***
+
+  doCreateUserWithEmailAndPassword = (email, password) =>
+    this.auth.createUserWithEmailAndPassword(email, password);
+
+  doSignInWithEmailAndPassword = (email, password) =>
+    this.auth.signInWithEmailAndPassword(email, password);
+
+  doSignOut = () => this.auth.signOut();
+
+  doPasswordReset = email => this.auth.sendPasswordResetEmail(email);
+
+  doPasswordUpdate = password =>
+    this.auth.currentUser.updatePassword(password);
+}
+
+export default Firebase;
+{{< /highlight >}}
+
+That's the authentication interface for your React components connected to the Firebase API. It covers all the use cases for this application. In the next section, we will consume all the methods of your Firebase class in your React components.
```

Comment:
"for your React components connected to the Firebase API" -> "for your React components eventually connected to the Firebase API"

Reply:
You don't really ever need to use the word eventually in English writing.  You can say 'that will do something' and the eventually is assumed in context, as eventually just means 'in the future', and all things that haven't happened yet are eventual.  I changed this to "React components that will connect to the Firebase API"